src/reader.cpp

```diff
@@ -3624,6 +3731,28 @@ void reader_data_t::handle_readline_command(readline_cmd_t c, readline_loop_stat
             }
             break;
         }
+        case rl::history_pager: {
+            if (history_pager_active) {
+                fill_history_pager(false, history_search_direction_t::backward);
+                break;
+            }
+
+            // Record our cycle_command_line.
+            cycle_command_line = command_line.text();
+            cycle_cursor_pos = command_line.position();
+
+            this->history_pager_active = true;
+            this->history_pager_history_index_start = 0;
+            this->history_pager_history_index_end = 0;
+            // Update the pager data.
+            pager.set_search_field_shown(true);
+            pager.set_prefix(L"$ ", false /* highlight */);
```

Comment:
One weird nitpick: We typically use ">" to mark the prompt - in our docs, in our default prompt, ...

We could use that here - or see if we have a multibyte locale and use `►`?

Reply:
`>` looks fine here, but is of course confusable for a redirection. On the other hand the `>` will all line up, so it's easy enough to ignore a column of `>`.

![Screenshot_20220728_213251](https://user-images.githubusercontent.com/5185367/181622403-73783a6b-29ee-4731-9f76-574b853e7130.png)

Reply:
sure, I like `>` better than `$` because it's smaller. Let's try the multibyte one too, we'll see if anyone complains

Reply:
Could try to highlight these as commands, I think might be the most recognizable way.

Reply:
Yep, I think that's the next thing on my list